### ISSUE TITLE
Allow user to specify AOI box filtering behavior in sliding window datasets

### ISSUE THREAD

**Issue by chrisjkuch** [azavea/raster-vision#1990]
## 🚀 Feature

Add the ability to include chips in sliding window datasets that are only partially contained by the AOI.

## Motivation

Currently, sliding window datasets only include a chip if the chip falls entirely within the specified AOI. It would be nice for complex AOI geometries to enable some overlap of chips with the AOI boundaries so that complete coverage within the AOI can be obtained.

## Pitch

This should be a pretty minimal change (add a kwarg to the sliding window dataset and pass it through to the existing box-within checking [here](https://github.com/azavea/raster-vision/blob/d17319401dad63251384083cd2b17f6481cc5380/rastervision_pytorch_learner/rastervision/pytorch_learner/dataset/dataset.py#L227-L228)) and a great quality of life improvement for me personally.

**Comment by AdeelH** [comment]
Agreed. I remember making this exact change a while ago for a specific project, but I guess it was never merged upstream.
